aten/src/ATen/native/TensorShape.cpp

```diff
@@ -87,6 +89,19 @@ Tensor& set_cpu_(Tensor& result) {
   return result;
 }
 
+// We can't re-use the cpu kernel here because we don't want to use the cpu allocator.
+Tensor& set_meta_(Tensor& result) {
+  caffe2::TypeMeta dtype = result.dtype();
+  Storage storage(
+      Storage::use_byte_size_t(),
+      0,
+      at::detail::GetMetaAllocator(),
```

Comment:
It looks like we don't register the meta allocator anywhere right now - but I should probably do that instead of making a new API just for meta. I'll do that.